src/fcc_bridge_node_mavsdk.cpp

```diff
@@ -0,0 +1,274 @@
+//
+// Created by Johan on 30.04.2024.
+//
+
+#include "fcc_bridge_node.hpp"
+
+// Libc header
+#include <cinttypes>
+#include <map>
+
+namespace fcc_bridge {
+namespace {
+// Path to uart device on Raspberry PI
+constexpr char const *UART_DEVICE_PATH = "/dev/serial0";
+
+// Baud rate to use for uart connection
+constexpr int UART_BAUD_RATE = 112500;
+
+// Simulator IP address
+constexpr char const *SIMULATOR_IP_ADDRESS = "127.0.0.1";
+// Simulator TCP Port number
+constexpr int SIMULATOR_PORT_NUMBER = 14540;
+
+// Map to get the MAVLink System id for the selected
+// {TEAM_ID : {SYS_ID, IS_SIMULATOR_TARGET}}
+const std::map<const std::string, std::pair<const u8, const bool>> SYS_ID_MAP{
+    {"UAV_TEAM_RED", {21, false}},
+    {"UAV_TEAM_GREEN", {22, false}},
+    {"UAV_TEAM_BLUE", {23, false}},
+    {"SIMULATOR", {1, true}}};
+
+// Timeout for the autopilot to be discovered
+constexpr double AUTOPILOT_DISCOVERY_TIMEOUT_S = 1;
+}  // namespace
+
+void FCCBridgeNode::setup_mavsdk() {
+    RCLCPP_DEBUG(this->get_logger(), "Setting up MAVSDK");
+
+    // Internal state check
+    if (this->internal_state != INTERNAL_STATE::ROS_SET_UP) {
+        RCLCPP_ERROR(this->get_logger(),
+                     "Repeated try to setup MAVSDK components");
+        this->internal_state = INTERNAL_STATE::ERROR;
+        return;
+    }
+    std::string uav_id = "__INVALID__";
+    try {
+        // Getting uav id to deduct MAVLink System ID
+        uav_id = this->declare_parameter<std::string>("UAV_ID");
+    } catch (
+        const rclcpp::exceptions::UninitializedStaticallyTypedParameterException
+            &e) {
+        // Catch if parameter was not set
+        RCLCPP_FATAL(this->get_logger(), "UAV_ID parameter was not set!");
+        this->internal_state = INTERNAL_STATE::ERROR;
+        return;
+    } catch (const rclcpp::exceptions::InvalidParameterTypeException &e) {
+        // Catch if parameter is not a string
+        RCLCPP_FATAL(
+            this->get_logger(),
+            "UAV_ID parameter was of wrong type! It has to be a string!");
+        this->internal_state = INTERNAL_STATE::ERROR;
+        return;
+    }
+
+    // Checking of for the supplied uav id there is an entry in the SYS_ID_MAP
```

Comment:
Weird grammar leading to bad readability. I suggest "Checking if there is an entry in the SYS_ID_MAP for the supplied uav id."

Reply:
Fixed in c0cdff7
I don´t know what I was thinking when writing that sentence.😅